src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,156 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
+    private static final int SIZE_EMPTY_LIST = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private Node<T> findByIndex(int index) {
+        Node<T> currentNode = head;
+        for (int i = 1; i <= index; i++) {
+            currentNode = currentNode.next;
+        }
+        return currentNode;
+    }
+
+    private Node<T> findByValue(T value) {
+        Node<T> currentNode = head;
+        while (currentNode != null) {
+            if (Objects.equals(currentNode.item, value)) {
+                return currentNode;
+            }
+            currentNode = currentNode.next;
+        }
+        return null;
+    }
+
+    private void removeNode(Node<T> node) {
+        if (node.prev != null) {
+            node.prev.next = node.next;
+        } else {
+            head = node.next;
+        }
+        if (node.next != null) {
+            node.next.prev = node.prev;
+        } else {
+            tail = node.prev;
+        }
+        size--;
+    }
+
+    private void checkIndexOutOfBoundsException(int index) {
+        if (index >= size() || index < SIZE_EMPTY_LIST) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
```

Comment:
move private fields to the end of your project

Reply:
moved only methods